### PR TITLE
fix: revamped convertors, added lazy class initialisations to Module, fixe…

### PR DESCRIPTION
**Pull request by RickSanchezStoic**
revamped convertors, added lazy class initialisations to Module, fixes pickling issues

### CODE REVIEW

**Review by vedpatwardhan — APPROVED** [review]
lgtm! Not sure why none of the converter tests got triggered in the [CI](https://github.com/unifyai/ivy/actions/runs/6050030361/job/16418585407?pr=22910#step:3:7), could you please run those locally and confirm if they all have the same result as on `main` before merging? Thanks @RickSanchezStoic 😄

**Comment by vedpatwardhan** [comment]
please don't merge before testing locally, seems like there's some [circular import](https://github.com/unifyai/ivy/actions/runs/6050030361/job/16418467269?pr=22910#step:3:186) errors to be fixed, thanks!

**Comment by RickSanchezStoic** [comment]
> please don't merge before testing locally, seems like there's some [circular import](https://github.com/unifyai/ivy/actions/runs/6050030361/job/16418467269?pr=22910#step:3:186) errors to be fixed, thanks!

oh yeah got that

**Pull request merged**
